### PR TITLE
Minor : Fix exception in search due to exception in database.displayN…

### PR DESCRIPTION
**Pull request by sonika-shah**
### Describe your changes:

Minor : Fix exception in search due to exception in database.displayName and databaseSchema.aggregation

#
### Type of change:

- [x] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/search/elasticsearch/ElasticSearchClient.java [hunk 1]
```diff
@@ -1297,6 +1297,23 @@ private static SearchSourceBuilder buildSearchAcrossIndexesBuilder(
...
     SearchSourceBuilder searchSourceBuilder = searchBuilder(queryBuilder, null, from, size);
+    searchSourceBuilder.aggregation(
+        AggregationBuilders.terms("database.name.keyword")
+            .field("database.name.keyword")
+            .size(MAX_AGGREGATE_SIZE));
+    searchSourceBuilder.aggregation(
+        AggregationBuilders.terms("databaseSchema.name.keyword")
+            .field("databaseSchema.name.keyword")
+            .size(MAX_AGGREGATE_SIZE));
+    return addAggregation(searchSourceBuilder);
```
Comment (TeddyCr) [inline openmetadata-service/src/main/java/org/openmetadata/service/search/elasticsearch/ElasticSearchClient.java:1308]: Should we do the aggregation on the fullyQualifiedName? Using name I believe if we have `service.database.schema` and `otherService.database.schema` then they'll both get aggregated under the same if we use `name` I believe.

File: openmetadata-service/src/main/java/org/openmetadata/service/search/opensearch/OpenSearchClient.java [hunk 2]
```diff
@@ -1282,6 +1282,24 @@ private static SearchSourceBuilder buildSearchAcrossIndexesBuilder(
...
     SearchSourceBuilder searchSourceBuilder = searchBuilder(queryBuilder, null, from, size);
+    searchSourceBuilder.aggregation(
+        AggregationBuilders.terms("database.name.keyword")
+            .field("database.name.keyword")
+            .size(MAX_AGGREGATE_SIZE));
+    searchSourceBuilder.aggregation(
+        AggregationBuilders.terms("databaseSchema.name.keyword")
+            .field("databaseSchema.name.keyword")
+            .size(MAX_AGGREGATE_SIZE));
+    return addAggregation(searchSourceBuilder);
```
Comment (TeddyCr) [inline openmetadata-service/src/main/java/org/openmetadata/service/search/opensearch/OpenSearchClient.java:1293]: Same comment here https://github.com/open-metadata/OpenMetadata/pull/18290/files#r1803006416

File: openmetadata-service/src/main/java/org/openmetadata/service/search/elasticsearch/ElasticSearchClient.java (hunk 1 above)
Comment (sonika-shah) [reply openmetadata-service/src/main/java/org/openmetadata/service/search/elasticsearch/ElasticSearchClient.java:1308]: hi @TeddyCr thanks for pointing your concern is valid, but for our usage we are not  dependent
only on a single aggregation while getting result, we check the condition with other aggregations as well
image

Comment (sonika-shah) [reply openmetadata-service/src/main/java/org/openmetadata/service/search/elasticsearch/ElasticSearchClient.java:1308]: have tried to replicate the same scenario with same database and schema, seems to be working fine.
Also will be removing the support for aggregations on name.keyword, as in most scenario we are moving to show results based on displayName

image

**Review by TeddyCr — APPROVED** [review]

**Review by harshach — APPROVED** [review]

**Pull request merged**